### PR TITLE
Notification panel update

### PR DESCRIPTION
**Pull request by winniedo**
Notification panel with newly defined model data

### CODE REVIEW

**Comment by janniclas** [comment]
The bell and the notification bubble look really nice, good work :) 
But when I run the code you provided in the pull request, no table is shown after clicking the bell button, as shown in the gif below. 
Further, I don't see any of the agreed changes in your code (the angular material table and the data model with the demo data we talked about 2 or 3 weeks ago).

screen recording 2018-10-05 at 15 14 19

**Comment by winniedo** [comment]
Thank you, Jan for your feedback,
In the module datatable-notif you can see the defined data model for the time a static table with the angular material. now I'm checking why it does not work, by me in local all is fine.

**Comment by janniclas** [comment]
the newly pushed content is not building for me, please have a look at this. 
image

File: client/src/app/dashboard/dashboard-card/dashboard-card.component.ts [hunk 1]
```diff
@@ -37,10 +37,14 @@ export class DashboardCardComponent implements OnInit {
   @Input() route: string;
   @Input() title: string;
   @Input() componentType: string;
+  @Input() collapse_target: string;
```
Comment (janniclas) [inline client/src/app/dashboard/dashboard-card/dashboard-card.component.ts]: please stick to camel case convention for variable and method names

File: client/src/app/dashboard/dashboard-overview/dashboard-overview.component.html [hunk 2]
```diff
@@ -19,18 +19,175 @@
 <div class="container-fluid">
   <div class="row">
     <div class="col-md-3 col-sm-4">
-    <app-dashboard-card route="crawler" componentType={{crawler}} title="Crawler" img="../assets/images/crawler.png"></app-dashboard-card>
+      <app-dashboard-card route="crawler" componentType={{crawler}} title="Crawler" img="../assets/images/crawler.png" collapse_target="#demo1" numberOfNotification="3"></app-dashboard-card>
```
Comment (janniclas) [inline client/src/app/dashboard/dashboard-overview/dashboard-overview.component.html]: please change the collapse targets ids to something meaningful. 
create a variable for the numberOfNotifications and use that one depending on the component type

File: client/src/app/dashboard/dashboard-overview/dashboard-overview.component.html [hunk 3]
```diff
@@ -19,18 +19,175 @@
...
     <div class="col-md-2 col-sm-4">
-    <app-statuscard></app-statuscard>
+      <app-statuscard></app-statuscard>
+    </div>
   </div>
+
+  <!--Begin notification pannels-->
+  <div id="demo1" class="collapse">
+    <h4>Crawler</h4>
+    <!--<table class="table table-striped">
```
Comment (janniclas) [inline client/src/app/dashboard/dashboard-overview/dashboard-overview.component.html]: please remove the uncommented code

File: client/src/app/dashboard/dashboard-overview/dashboard-overview.component.html [hunk 4]
```diff
@@ -19,18 +19,175 @@
...
     <div class="col-md-1 col-sm-4">
     </div>
     <div class="col-md-2 col-sm-4">
-    <app-statuscard></app-statuscard>
+      <app-statuscard></app-statuscard>
+    </div>
   </div>
+
+  <!--Begin notification pannels-->
+  <div id="demo1" class="collapse">
```
Comment (janniclas) [inline client/src/app/dashboard/dashboard-overview/dashboard-overview.component.html]: please use a more angular like way of doing this. best would most likely be to use a ng-if (https://angular.io/api/common/NgIf)

File: client/src/app/dashboard/datatable-notif/datatable-notif-datasource.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
+import { DataSource } from '@angular/cdk/collections';
+import { MatPaginator, MatSort } from '@angular/material';
+import { map } from 'rxjs/operators';
+import { Observable, of as observableOf, merge } from 'rxjs';
+
+
+//Data model type
+export interface DatatableNotifItem {
+  type: string;
+  notif_name: string;
```
Comment (janniclas) [inline client/src/app/dashboard/datatable-notif/datatable-notif-datasource.ts]: naming convention camel case

File: client/src/app/dashboard/datatable-notif/datatable-notif.component.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { MatPaginator, MatSort } from '@angular/material';
+import { DatatableNotifDataSource } from './datatable-notif-datasource';
+
+@Component({
+  selector: 'app-datatable-notif',
+  templateUrl: './datatable-notif.component.html',
+  styleUrls: ['./datatable-notif.component.css']
+})
+export class DatatableNotifComponent implements OnInit {
```
Comment (janniclas) [inline client/src/app/dashboard/datatable-notif/datatable-notif.component.ts]: this class somehow needs to know for which component it needs to receive the notifications. a type variable might be necessary here.

**Comment by janniclas** [comment]
also the test case regarding the DatatableNotifComponent is failing, please have a look at that as well

**Comment by janniclas** [comment]
and the linter reports a lot of codestyle issues in the datatable-notif components, please resolve those

**Comment by janniclas** [comment]
@windor12 any further progress regarding the review? The way I see it right now the current version of the branch, after your latest commit, doesn't contain the requested changes.

**Pull request closed without merging**